File: packages/web/src/api/comment.api.ts
import { createApi } from '@reduxjs/toolkit/query/react'

import baseQueryWithReauth from './baseQueryWithReauth'

import { CommentDto, ICommentResponse } from '~types/comment/comment.type'

export const api = createApi({
  reducerPath: 'commentApi',
  baseQuery: baseQueryWithReauth,
  endpoints: builder => ({
    getComment: builder.query<ICommentResponse, { id: string }>({
      query: ({ id }) => ({
        url: `comments/${id}`,
        method: 'GET',
      }),
    }),
    deleteCommentById: builder.mutation<{ message: string }, { id: string }>({
      query: ({ id }) => ({
        url: `comments/${id}`,
        method: 'DELETE',
      }),
    }),
    createComment: builder.mutation<ICommentResponse, CommentDto>({
      query: body => ({
        url: 'comments',
        method: 'POST',
        body,
      }),
    }),
  }),
})

export const { useCreateCommentMutation, useLazyGetCommentQuery, useDeleteCommentByIdMutation } = api

File: packages/mobile/app/src/main/java/com/devhub/devhubapp/api/CommentAPI.kt
package com.devhub.devhubapp.api

import com.devhub.devhubapp.dataClasses.AddCommentRequest
import com.devhub.devhubapp.dataClasses.Comment
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.POST
import retrofit2.http.Path

interface CommentAPI {
    @POST("comments")
    fun createComment(
        @Body commentRequest: AddCommentRequest
    ): Call<Comment>

    @DELETE("comments/{id}")
    fun deleteComment(
        @Path("id") commentId: String
    ): Call<Unit>
}